aten/src/THCUNN/generic/MultiLabelMarginCriterion.cu

```diff
@@ -149,26 +152,29 @@ void THNN_(MultiLabelMarginCriterion_updateGradInput)(
   {
     int nframe = gradInput->size(0);
     int dim = gradInput->size(1);
-    THArgCheck(!target->is_empty() && (target->dim() == 2) && (target->size(0) == nframe)
+    THArgCheck((input->size(1) != 0) && (target->dim() == 2) && (target->size(0) == nframe)
                && (target->size(1) == dim), 3, "inconsistent target size");
-    THArgCheck(!istarget->is_empty() && (istarget->dim() == 2) && (istarget->size(0) == nframe)
+    THArgCheck((istarget->dim() == 2) && (istarget->size(0) == nframe)
                && (istarget->size(1) == dim), 3, "inconsistent isTarget size");
-    dim3 blocks(gradInput->size(0));
-    dim3 threads(MULTILABELMARGIN_THREADS);
 
-    cunn_MultiLabelMarginCriterion_updateGradInput_kernel<scalar_t, accreal>
-      <<<blocks, threads, 0, c10::cuda::getCurrentCUDAStream()>>>(
-        THCTensor_(data)(state, gradInput),
-        THCTensor_(data)(state, gradOutput),
-        THCTensor_(data)(state, input),
-        THCIndexTensor_(data)(state, target),
-        THCTensor_(data)(state, istarget),
-        gradInput->size(0), gradInput->size(1),
-        reduction == at::Reduction::Mean,
-        reduction != at::Reduction::None);
+    if (input->numel() != 0) {
```

Comment:
Seems like this check should also live at the start of the function and short-circuit earlier.

Reply:
Done.